### PR TITLE
Integrate openfisca-uk-data

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
### Dataset generation logic

- [x] Raw FRS 
- [x] Raw LCFS
- [x] Raw WAS
- [x] Raw SPI
- [x] FRS (original)
- [x] FRS (enhanced)
- [x] SPI (original)

### CODE REVIEW

**Comment by nikhilwoodruff** [comment]
As per our discussion I'll also ensure the uprating for LCFS and WAS imputations (LCFS from 2020* and WAS from 2018-2020) are complete.

For the LCFS, I found a breakdown of the CPI by consumption category, so I've added it as a parameter.

**Comment by nikhilwoodruff** [comment]
Got the LCFS and WAS uprating fully in, just stuck on this error after trying to load the Extended FRS using 2022:
```console
ValueError: Input [    0.          5084.08007812     0.         ...     0.
 25140.             0.        ] is not a valid value for the entity benunit (size = 44034 != 45466 = count)
 ```

**Comment by nikhilwoodruff** [comment]
OK, able to fix that - seemed to be an issue with using NumPy to clone the benunit unit IDs.

**Comment by nikhilwoodruff** [comment]
Fixed a particularly elusive bug: I couldn't figure out why the calibration process wasn't making a dent in the budgetary impacts: turned out to be that state pension income got eliminated by the extension process when duplicating it (due to similarity for `state_id` and `state_role` that need to be treated differently), resulting in a massive pension credit (£30bn) expenditure which the optimizer couldn't do much to fix.

File: openfisca_uk/data/datasets/frs/enhanced/stages/extension/extended_frs.py [hunk 1]
```diff
@@ -66,8 +66,8 @@ def generate(self, year: int):
             year, 
             {
                 **uc_migrated,
-                f"in_original_frs/{year}": [False] * NUM_HOUSEHOLDS,
-                f"uc_migrated/{year}": [True] * NUM_HOUSEHOLDS,
+                f"in_original_frs/{year}": [False] * NUM_HOUSEHOLDS * 2,
```
Comment (MaxGhenis) [inline openfisca_uk/data/datasets/frs/enhanced/stages/extension/extended_frs.py]: could you change this to `HOUSEHOLD_COUNT`?

**Comment by nikhilwoodruff** [comment]
@MaxGhenis I'm actually leaving https://github.com/PolicyEngine/openfisca-uk/issues/571 out of this - it'd need a fairly significant rewrite of a lot of the core logic (designed at the moment to apply weights at the same place as the parameter-related operations).

**Comment by nikhilwoodruff** [comment]
Calibration finally done!

- [x] Extended FRS
- [x] Calibrated FRS
- [x] Enhanced FRS

File: changelog.yaml [hunk 2]
```diff
@@ -395,3 +395,7 @@
     fixed:
     - Chart on carbon intensities.
   date: 2022-04-08 10:35:59
+- bump: major
```
Comment (MaxGhenis) [inline changelog.yaml]: This is definitely a huge release, but I had some backward-incompatible changes in mind before moving to 1.0, like lowercasing variable names and reorganizing parameter trees around agencies.

Comment (nikhilwoodruff) [reply changelog.yaml]: Yeah agreed - sticking to minor

**Pull request merged**